frontends/web/src/routes/device/bitbox02/bitbox02.tsx

```diff
@@ -325,39 +325,38 @@ class BitBox02 extends Component<Props, State> {
   };
 
   private setDeviceName = (event: FormEvent) => {
+    const { deviceID, t } = this.props;
     event.preventDefault();
     this.setState({
-      waitDialog: { title: this.props.t('bitbox02Interact.confirmName') }
-    }, () => {
-      setDeviceName(this.props.deviceID, this.state.deviceName)
-        .then(() => {
-          this.setState(
-            { waitDialog: undefined },
-            () => this.setPassword(),
-          );
-        })
-        .catch(result => {
-          if (result.message) {
-            alertUser(result.message, {
-              asDialog: false,
-              callback: () => this.setState({ waitDialog: undefined }),
-            });
-          }
+      waitDialog: { title: t('bitbox02Interact.confirmName') }
+    }, async () => {
+      const result = await setDeviceName(deviceID, this.state.deviceName);
+      if (!result.success) {
+        alertUser(result.message || t('genericError'), {
```

Comment:
added genericError as fallback since typescript wasn't happy, as result.message is optional.

before it checked if result.message is truthy, but if missing it didn't show any error…

not tested yet

Reply:
tested and works fine.